backend/src/Database.hs

```diff
@@ -66,20 +76,33 @@ getMostRecentStats pid =
       logErrorN $ "Missing stats for PID " <>: pid <> ". Inserting defaults."
       year <- currentYear
       let stats = defaultPlayerStats pid year
-      SQL.insert_ stats
+      insert_ stats
       pure stats
   where
-    recentStats = SQL.selectOne $ do
+    recentStats = selectOne $ do
       stats <- from $ table @PlayerStats
       where_ (stats ^. PlayerStatsPlayerId ==. val pid)
       orderBy [desc (stats ^. PlayerStatsYear)]
       pure stats
 
 replacePlayerStats :: (MonadIO m, MonadLogger m) => PlayerStats -> SqlPersistT m ()
-replacePlayerStats stats@(PlayerStats {..}) = SQL.replace (PlayerStatsKey playerStatsPlayerId playerStatsYear) stats
+replacePlayerStats stats@(PlayerStats {..}) = replace (PlayerStatsKey playerStatsPlayerId playerStatsYear) stats
 
 insertRatingChange :: (MonadIO m, MonadLogger m) => RatingDiff -> SqlPersistT m ()
-insertRatingChange = SQL.insert_
+insertRatingChange = insert_
 
 insertGameReport :: (MonadIO m, MonadLogger m) => GameReport -> SqlPersistT m (Key GameReport)
-insertGameReport = SQL.insert
+insertGameReport = insert
+
+getGameReports :: (MonadIO m, MonadLogger m) => SqlPersistT m [(Entity GameReport, Entity Player, Entity Player)]
+getGameReports =
+  select $ do
+    (report :& winner :& loser) <-
+      from $
+        table @GameReport
+          `innerJoin` table @Player
+            `on` (\(report :& winner) -> report ^. GameReportWinnerId ==. winner ^. PlayerId)
+          `innerJoin` table @Player
+            `on` (\(report :& _ :& loser) -> report ^. GameReportLoserId ==. loser ^. PlayerId)
+    orderBy [desc (report ^. GameReportTimestamp)]
```

Comment:
Much hoogling of new symbols here for me haha. `^.` is particularly helpful ✨

Reply:
You don't want to know how much time I spent trying to debug why this query wouldn't compile, only to eventually realize I was using `on` from `Prelude` instead of `on` from `esqueleto`

Reply:
Noooooo haha that's awful